engine/flink/management/sample/src/main/scala/pl/touk/nussknacker/engine/management/sample/source/GenericSourceWithCustomVariablesSample.scala

```diff
@@ -60,13 +56,16 @@ object GenericSourceWithCustomVariablesSample extends SourceFactory[String] with
   //There is only one parameter in this source
   private val elementsParamName = "elements"
 
-  private val customContextInitializer: BasicFlinkGenericContextInitializer[String, DefinedParameter] = new CustomFlinkContextInitializer
+  private val customContextInitializer: ContextInitializer[String] = new CustomFlinkContextInitializer
 
   override def contextTransformation(context: ValidationContext, dependencies: List[NodeDependencyValue])(implicit nodeId: ProcessCompilationError.NodeId)
   : GenericSourceWithCustomVariablesSample.NodeTransformationDefinition = {
     case TransformationStep(Nil, _) => NextParameters(Parameter[java.util.List[String]](`elementsParamName`) :: Nil)
     case step@TransformationStep((`elementsParamName`, _) :: Nil, None) =>
-      FinalResults(customContextInitializer.validationContext(context, dependencies, step.parameters))
+      val validContextAfterInitialization = customContextInitializer.validationContext(context)
+      FinalResults(
+        validContextAfterInitialization.getOrElse(context),
+        validContextAfterInitialization.swap.map(_.toList).getOrElse(Nil))
```

Comment:
I think we convert result of contextInitializer to FinalResults in the same way many times, maybe we can extract some helper or method?

Reply:
Added `FinalResults.forValidation` in separate commit - take a look